packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1042,11 +1042,28 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
   /// The given [BuildContext] will be rebuilt if the state of the route changes
   /// while it is visible (specifically, if [isCurrent] or [canPop] change value).
   @optionalTypeArgs
-  static ModalRoute<T>? of<T extends Object?>(BuildContext context) {
-    final _ModalScopeStatus? widget = context.dependOnInheritedWidgetOfExactType<_ModalScopeStatus>();
+  static ModalRoute<T>? of<T extends Object?>(
+    BuildContext context, {
+    bool createDependency = true,
+  }) {
+    _ModalScopeStatus? widget;
+    if (createDependency) {
+      widget = context.dependOnInheritedWidgetOfExactType<_ModalScopeStatus>();
+    } else {
+      widget = context
+          .getElementForInheritedWidgetOfExactType<_ModalScopeStatus>()
+          ?.widget as _ModalScopeStatus?;
+    }
     return widget?.route as ModalRoute<T>?;
   }
 
+  /// Obtains the element corresponding to the nearest widget of
+  /// [_ModalScopeStatus] within the given [context].
+  @visibleForTesting
```

Comment:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-visiblefortesting

Reply:
Added @protected

Reply:
It's a static method. `@protected` doesn't make sense here.